Add better instructions for how to run lint and pre commit

@njbrake Running mypy gives me a lot of errors:
```
UV_SYSTEM_PYTHON=0 uv run pre-commit run --all-files
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Using pre-commit with uv 0.11.0 via pre-commit-uv 4.2.1
[INFO] Installing environment for https://github.com/astral-sh/ruff-pre-commit.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Using pre-commit with uv 0.11.0 via pre-commit-uv 4.2.1
[INFO] Installing environment for https://github.com/codespell-project/codespell.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Using pre-commit with uv 0.11.0 via pre-commit-uv 4.2.1
[INFO] Installing environment for https://github.com/kynan/nbstripout.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Using pre-commit with uv 0.11.0 via pre-commit-uv 4.2.1
check for added large files..............................................Passed
check for case conflicts.................................................Passed
check json...............................................................Passed
check for merge conflicts................................................Passed
check toml...............................................................Passed
check yaml...............................................................Passed
fix end of files.........................................................Passed
mixed line ending........................................................Passed
sort simple yaml files...............................(no files to check)Skipped
trim trailing whitespace.................................................Passed
ruff (legacy alias)......................................................Passed
ruff format..............................................................Passed
mypy.....................................................................Failed
- hook id: mypy
- duration: 7.74s
- exit code: 1

src/any_llm/gateway/core/database.py:5: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
src/any_llm/gateway/core/database.py:6: error: Cannot find implementation or library stub for module named "alembic.config"  [import-not-found]
src/any_llm/gateway/core/database.py:7: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
src/any_llm/gateway/core/database.py:8: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/core/database.py:31: error: Untyped decorator makes function "_set_sqlite_pragma" untyped  [untyped-decorator]
src/any_llm/gateway/alembic/versions/1e382aa3a9e7_add_per_user_budget_resets.py:11: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
src/any_llm/gateway/alembic/versions/1e382aa3a9e7_add_per_user_budget_resets.py:12: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
tests/gateway/test_models_endpoint.py:5: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_pagination_bounds.py:3: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
src/any_llm/gateway/alembic/versions/e7c85cc73bfa_convert_budget_duration_to_seconds.py:11: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
src/any_llm/gateway/alembic/versions/e7c85cc73bfa_convert_budget_duration_to_seconds.py:12: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
src/any_llm/gateway/services/pricing_service.py:3: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/services/pricing_service.py:25: error: Returning Any from function declared to return "ModelPricing | None"  [no-any-return]
tests/gateway/test_find_model_pricing.py:3: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/api/deps.py:5: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/api/deps.py:6: error: Cannot find implementation or library stub for module named "sqlalchemy.exc"  [import-not-found]
src/any_llm/gateway/api/deps.py:7: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/api/deps.py:54: error: Returning Any from function declared to return "str"  [no-any-return]
src/any_llm/gateway/api/deps.py:59: error: Returning Any from function declared to return "str"  [no-any-return]
src/any_llm/gateway/api/deps.py:108: error: Returning Any from function declared to return "APIKey"  [no-any-return]
src/any_llm/gateway/api/routes/models.py:6: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/api/routes/models.py:8: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/api/routes/models.py:44: error: Untyped decorator makes function "list_models" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/models.py:57: error: Untyped decorator makes function "get_model" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/keys.py:5: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/api/routes/keys.py:7: error: Cannot find implementation or library stub for module named "sqlalchemy.exc"  [import-not-found]
src/any_llm/gateway/api/routes/keys.py:7: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
src/any_llm/gateway/api/routes/keys.py:8: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/api/routes/keys.py:74: error: Untyped decorator makes function "create_key" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/keys.py:139: error: Untyped decorator makes function "list_keys" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/keys.py:154: error: Untyped decorator makes function "get_key" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/keys.py:174: error: Untyped decorator makes function "update_key" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/keys.py:214: error: Untyped decorator makes function "delete_key" untyped  [untyped-decorator]
tests/unit/test_gateway_metrics.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
tests/unit/test_gateway_metrics.py:5: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
src/any_llm/providers/voyage/utils.py:4: error: Cannot find implementation or library stub for module named "voyageai.object.embeddings"  [import-not-found]
src/any_llm/providers/voyage/utils.py:12: error: Unused "type: ignore" comment  [unused-ignore]
tests/gateway/test_messages_endpoint.py:7: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
src/any_llm/providers/azure/utils.py:4: error: Cannot find implementation or library stub for module named "azure.ai.inference.models"  [import-not-found]
src/any_llm/providers/xai/utils.py:5: error: Cannot find implementation or library stub for module named "xai_sdk.chat"  [import-not-found]
src/any_llm/providers/xai/utils.py:8: error: Cannot find implementation or library stub for module named "xai_sdk.proto"  [import-not-found]
src/any_llm/providers/xai/utils.py:78: error: Unused "type: ignore" comment  [unused-ignore]
src/any_llm/gateway/services/budget_service.py:3: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/services/budget_service.py:4: error: Cannot find implementation or library stub for module named "sqlalchemy.exc"  [import-not-found]
src/any_llm/gateway/services/budget_service.py:5: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/services/budget_service.py:132: error: Returning Any from function declared to return "bool"  [no-any-return]
src/any_llm/providers/gemini/base.py:26: error: Cannot find implementation or library stub for module named "google.genai"  [import-not-found]
src/any_llm/providers/gemini/base.py:43: error: Cannot find implementation or library stub for module named "google"  [import-not-found]
src/any_llm/providers/xai/xai.py:14: error: Cannot find implementation or library stub for module named "xai_sdk"  [import-not-found]
src/any_llm/providers/xai/xai.py:15: error: Cannot find implementation or library stub for module named "xai_sdk.chat"  [import-not-found]
src/any_llm/providers/xai/xai.py:150: error: Cannot find implementation or library stub for module named "xai_sdk.proto"  [import-not-found]
src/any_llm/providers/xai/xai.py:176: error: Unused "type: ignore" comment  [unused-ignore]
src/any_llm/providers/ollama/ollama.py:13: error: Cannot find implementation or library stub for module named "ollama"  [import-not-found]
src/any_llm/providers/watsonx/watsonx.py:22: error: Cannot find implementation or library stub for module named "ibm_watsonx_ai"  [import-not-found]
src/any_llm/providers/watsonx/watsonx.py:24: error: Cannot find implementation or library stub for module named "ibm_watsonx_ai.foundation_models.inference.model_inference"  [import-not-found]
src/any_llm/providers/mistral/mistral.py:15: error: Cannot find implementation or library stub for module named "mistralai"  [import-not-found]
src/any_llm/providers/mistral/mistral.py:16: error: Cannot find implementation or library stub for module named "mistralai.extra"  [import-not-found]
src/any_llm/providers/mistral/mistral.py:17: error: Cannot find implementation or library stub for module named "mistralai.models.responseformat"  [import-not-found]
src/any_llm/providers/mistral/mistral.py:37: error: Cannot find implementation or library stub for module named "mistralai.models"  [import-not-found]
src/any_llm/providers/mistral/mistral.py:38: error: Cannot find implementation or library stub for module named "mistralai.models.embeddingresponse"  [import-not-found]
src/any_llm/providers/mistral/mistral.py:145: error: Unused "type: ignore" comment  [unused-ignore]
tests/unit/providers/test_mistral_provider.py:379: error: Cannot find implementation or library stub for module named "mistralai.types.basemodel"  [import-not-found]
src/any_llm/providers/voyage/voyage.py:21: error: Cannot find implementation or library stub for module named "voyageai.client_async"  [import-not-found]
src/any_llm/gateway/services/pricing_init_service.py:3: error: Cannot find implementation or library stub for module named "sqlalchemy.exc"  [import-not-found]
src/any_llm/gateway/services/pricing_init_service.py:4: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/providers/groq/groq.py:20: error: Cannot find implementation or library stub for module named "groq"  [import-not-found]
src/any_llm/providers/groq/groq.py:34: error: Cannot find implementation or library stub for module named "groq.types.chat"  [import-not-found]
src/any_llm/providers/groq/groq.py:34: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
tests/unit/providers/test_groq_provider.py:113: error: Cannot find implementation or library stub for module named "groq.types.chat"  [import-not-found]
src/any_llm/providers/huggingface/huggingface.py:26: error: Cannot find implementation or library stub for module named "huggingface_hub"  [import-not-found]
src/any_llm/providers/huggingface/huggingface.py:42: error: Unused "type: ignore" comment  [unused-ignore]
src/any_llm/providers/huggingface/huggingface.py:42: error: Cannot find implementation or library stub for module named "huggingface_hub.inference._generated.types"  [import-not-found]
src/any_llm/providers/huggingface/huggingface.py:42: note: Error code "import-not-found" not covered by "type: ignore" comment
src/any_llm/providers/platform/utils.py:10: error: Cannot find implementation or library stub for module named "any_llm_platform_client"  [import-not-found]
src/any_llm/providers/platform/utils.py:13: error: Cannot find implementation or library stub for module named "opentelemetry"  [import-not-found]
src/any_llm/providers/platform/utils.py:15: error: Cannot find implementation or library stub for module named "opentelemetry.exporter.otlp.proto.http.trace_exporter"  [import-not-found]
src/any_llm/providers/platform/utils.py:16: error: Cannot find implementation or library stub for module named "opentelemetry.sdk.resources"  [import-not-found]
src/any_llm/providers/platform/utils.py:17: error: Cannot find implementation or library stub for module named "opentelemetry.sdk.trace"  [import-not-found]
src/any_llm/providers/platform/utils.py:18: error: Cannot find implementation or library stub for module named "opentelemetry.sdk.trace.export"  [import-not-found]
src/any_llm/providers/platform/utils.py:19: error: Cannot find implementation or library stub for module named "opentelemetry.trace"  [import-not-found]
src/any_llm/providers/platform/utils.py:28: error: Cannot find implementation or library stub for module named "opentelemetry.context"  [import-not-found]
src/any_llm/providers/platform/utils.py:184: error: Class cannot subclass "SpanProcessor" (has type "Any")  [misc]
src/any_llm/providers/platform/utils.py:209: error: Unused "type: ignore" comment  [unused-ignore]
src/any_llm/providers/platform/utils.py:299: error: Unused "type: ignore" comment  [unused-ignore]
src/any_llm/gateway/api/routes/health.py:3: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/api/routes/health.py:4: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
src/any_llm/gateway/api/routes/health.py:13: error: Untyped decorator makes function "health_check" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/health.py:23: error: Untyped decorator makes function "health_liveness" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/health.py:37: error: Untyped decorator makes function "health_readiness" untyped  [untyped-decorator]
tests/gateway/test_timezone_consistency.py:4: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/providers/gemini/gemini.py:4: error: Cannot find implementation or library stub for module named "google"  [import-not-found]
src/any_llm/providers/gemini/gemini.py:5: error: Cannot find implementation or library stub for module named "google.genai"  [import-not-found]
src/any_llm/gateway/api/main.py:1: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
tests/unit/providers/test_platform_provider.py:10: error: Cannot find implementation or library stub for module named "any_llm_platform_client"  [import-not-found]
tests/unit/providers/test_platform_provider.py:11: error: Cannot find implementation or library stub for module named "opentelemetry.sdk.trace"  [import-not-found]
tests/unit/providers/test_platform_provider.py:12: error: Cannot find implementation or library stub for module named "opentelemetry.sdk.trace.export"  [import-not-found]
tests/unit/providers/test_platform_provider.py:13: error: Cannot find implementation or library stub for module named "opentelemetry.sdk.trace.export.in_memory_span_exporter"  [import-not-found]
tests/unit/providers/test_platform_provider.py:1774: error: Cannot find implementation or library stub for module named "opentelemetry"  [import-not-found]
tests/unit/test_gateway_root_page.py:3: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/conftest.py:11: error: Cannot find implementation or library stub for module named "uvicorn"  [import-not-found]
tests/gateway/conftest.py:12: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
tests/gateway/conftest.py:13: error: Cannot find implementation or library stub for module named "alembic.config"  [import-not-found]
tests/gateway/conftest.py:14: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/conftest.py:15: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
tests/gateway/conftest.py:16: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/gateway/test_key_management.py:4: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_client_args.py:6: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_client_args.py:7: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
tests/gateway/test_client_args.py:8: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
Found 106 errors in 37 files (checked 75 source files)
src/any_llm/gateway/core/database.py:5: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
src/any_llm/gateway/core/database.py:6: error: Cannot find implementation or library stub for module named "alembic.config"  [import-not-found]
src/any_llm/gateway/core/database.py:7: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
src/any_llm/gateway/core/database.py:8: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/core/database.py:31: error: Untyped decorator makes function "_set_sqlite_pragma" untyped  [untyped-decorator]
tests/gateway/test_health.py:1: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
src/any_llm/gateway/alembic/versions/967575f779b7_add_compound_index_usage_logs_user_id_timestamp.py:11: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
src/any_llm/gateway/models/entities.py:5: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
src/any_llm/gateway/models/entities.py:6: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/models/entities.py:9: error: Class cannot subclass "DeclarativeBase" (has type "Any")  [misc]
src/any_llm/gateway/alembic/env.py:5: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
src/any_llm/gateway/alembic/env.py:6: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
src/any_llm/gateway/core/config.py:8: error: Cannot find implementation or library stub for module named "pydantic_settings"  [import-not-found]
src/any_llm/gateway/core/config.py:8: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
src/any_llm/gateway/core/config.py:20: error: Class cannot subclass "BaseSettings" (has type "Any")  [misc]
tests/gateway/test_user_budget.py:3: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/unit/providers/test_cohere_exceptions.py:8: error: Cannot find implementation or library stub for module named "cohere"  [import-not-found]
src/any_llm/providers/voyage/utils.py:4: error: Cannot find implementation or library stub for module named "voyageai.object.embeddings"  [import-not-found]
src/any_llm/providers/voyage/utils.py:12: error: Unused "type: ignore" comment  [unused-ignore]
tests/unit/providers/test_gemini_exceptions.py:8: error: Cannot find implementation or library stub for module named "google.genai.errors"  [import-not-found]
src/any_llm/providers/huggingface/utils.py:5: error: Cannot find implementation or library stub for module named "huggingface_hub.hf_api"  [import-not-found]
src/any_llm/providers/huggingface/utils.py:6: error: Unused "type: ignore" comment  [unused-ignore]
src/any_llm/providers/huggingface/utils.py:6: error: Cannot find implementation or library stub for module named "huggingface_hub.inference._generated.types"  [import-not-found]
src/any_llm/providers/huggingface/utils.py:6: note: Error code "import-not-found" not covered by "type: ignore" comment
src/any_llm/providers/ollama/utils.py:6: error: Cannot find implementation or library stub for module named "ollama"  [import-not-found]
src/any_llm/providers/mistral/utils.py:5: error: Cannot find implementation or library stub for module named "mistralai.models"  [import-not-found]
src/any_llm/providers/mistral/utils.py:11: error: Cannot find implementation or library stub for module named "mistralai.models.chatcompletionresponse"  [import-not-found]
src/any_llm/providers/mistral/utils.py:12: error: Cannot find implementation or library stub for module named "mistralai.models.toolcall"  [import-not-found]
src/any_llm/providers/mistral/utils.py:13: error: Cannot find implementation or library stub for module named "mistralai.types.basemodel"  [import-not-found]
src/any_llm/providers/mistral/utils.py:42: error: Cannot find implementation or library stub for module named "mistralai.models.embeddingresponse"  [import-not-found]
src/any_llm/providers/mistral/utils.py:301: error: Unused "type: ignore" comment  [unused-ignore]
src/any_llm/providers/mistral/utils.py:396: error: Unused "type: ignore" comment  [unused-ignore]
src/any_llm/providers/groq/utils.py:3: error: Cannot find implementation or library stub for module named "groq.types"  [import-not-found]
src/any_llm/providers/groq/utils.py:4: error: Cannot find implementation or library stub for module named "groq.types.chat"  [import-not-found]
src/any_llm/gateway/api/routes/_helpers.py:6: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/providers/cohere/cohere.py:13: error: Cannot find implementation or library stub for module named "cohere"  [import-not-found]
src/any_llm/providers/cohere/cohere.py:110: error: Unused "type: ignore" comment  [unused-ignore]
src/any_llm/providers/cohere/cohere.py:154: error: Unused "type: ignore" comment  [unused-ignore]
tests/gateway/test_shared_helpers.py:7: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
tests/gateway/test_shared_helpers.py:8: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/unit/providers/test_ollama_provider.py:4: error: Cannot find implementation or library stub for module named "ollama"  [import-not-found]
tests/unit/providers/test_azure_provider.py:6: error: Cannot find implementation or library stub for module named "azure.ai.inference.models"  [import-not-found]
src/any_llm/gateway/api/routes/chat.py:6: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/api/routes/chat.py:7: error: Cannot find implementation or library stub for module named "fastapi.responses"  [import-not-found]
src/any_llm/gateway/api/routes/chat.py:9: error: Cannot find implementation or library stub for module named "sqlalchemy.exc"  [import-not-found]
src/any_llm/gateway/api/routes/chat.py:10: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/api/routes/chat.py:174: error: Untyped decorator makes function "chat_completions" untyped  [untyped-decorator]
src/any_llm/providers/platform/platform.py:6: error: Cannot find implementation or library stub for module named "any_llm_platform_client"  [import-not-found]
src/any_llm/providers/platform/platform.py:8: error: Cannot find implementation or library stub for module named "opentelemetry"  [import-not-found]
src/any_llm/providers/platform/platform.py:9: error: Cannot find implementation or library stub for module named "opentelemetry.trace"  [import-not-found]
src/any_llm/gateway/api/routes/messages.py:3: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/api/routes/messages.py:4: error: Cannot find implementation or library stub for module named "fastapi.responses"  [import-not-found]
src/any_llm/gateway/api/routes/messages.py:4: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
src/any_llm/gateway/api/routes/messages.py:6: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/api/routes/messages.py:64: error: Untyped decorator makes function "create_message" untyped  [untyped-decorator]
src/any_llm/providers/vertexai/vertexai.py:3: error: Cannot find implementation or library stub for module named "google.genai"  [import-not-found]
src/any_llm/providers/vertexai/vertexai.py:3: error: Cannot find implementation or library stub for module named "google"  [import-not-found]
tests/integration/test_cached_tokens.py:14: error: Cannot find implementation or library stub for module named "google.genai"  [import-not-found]
tests/integration/test_cached_tokens.py:14: error: Cannot find implementation or library stub for module named "google"  [import-not-found]
tests/integration/test_cached_tokens.py:15: error: Cannot find implementation or library stub for module named "google.genai.types"  [import-not-found]
tests/unit/providers/test_vertexai_provider.py:3: error: Cannot find implementation or library stub for module named "google.genai.types"  [import-not-found]
tests/unit/providers/test_vertexai_provider.py:3: error: Cannot find implementation or library stub for module named "google.genai"  [import-not-found]
tests/gateway/conftest.py:11: error: Cannot find implementation or library stub for module named "uvicorn"  [import-not-found]
tests/gateway/conftest.py:12: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
tests/gateway/conftest.py:13: error: Cannot find implementation or library stub for module named "alembic.config"  [import-not-found]
tests/gateway/conftest.py:14: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/conftest.py:15: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
tests/gateway/conftest.py:16: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/gateway/test_user_delete_preserve_logs.py:6: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_user_delete_preserve_logs.py:7: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
tests/gateway/test_user_delete_preserve_logs.py:8: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/gateway/test_gateway_completion.py:6: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
Found 69 errors in 28 files (checked 75 source files)
src/any_llm/gateway/core/database.py:5: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
src/any_llm/gateway/core/database.py:6: error: Cannot find implementation or library stub for module named "alembic.config"  [import-not-found]
src/any_llm/gateway/core/database.py:7: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
src/any_llm/gateway/core/database.py:8: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/core/database.py:31: error: Untyped decorator makes function "_set_sqlite_pragma" untyped  [untyped-decorator]
tests/gateway/test_error_detail_leakage.py:5: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_streaming_error_event.py:5: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_message_validation.py:3: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
src/any_llm/gateway/services/pricing_service.py:3: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/services/pricing_service.py:25: error: Returning Any from function declared to return "ModelPricing | None"  [no-any-return]
src/any_llm/gateway/metrics.py:8: error: Cannot find implementation or library stub for module named "prometheus_client"  [import-not-found]
src/any_llm/gateway/metrics.py:9: error: Cannot find implementation or library stub for module named "starlette.responses"  [import-not-found]
src/any_llm/gateway/metrics.py:12: error: Cannot find implementation or library stub for module named "starlette.requests"  [import-not-found]
src/any_llm/gateway/metrics.py:13: error: Cannot find implementation or library stub for module named "starlette.types"  [import-not-found]
src/any_llm/gateway/auth/vertex_auth.py:9: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/auth/vertex_auth.py:9: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
tests/unit/providers/test_mistral_exceptions.py:9: error: Cannot find implementation or library stub for module named "mistralai.models"  [import-not-found]
tests/unit/providers/test_mistral_exceptions.py:41: error: Cannot find implementation or library stub for module named "mistralai.models.httpvalidationerror"  [import-not-found]
src/any_llm/providers/voyage/utils.py:4: error: Cannot find implementation or library stub for module named "voyageai.object.embeddings"  [import-not-found]
src/any_llm/providers/voyage/utils.py:12: error: Unused "type: ignore" comment  [unused-ignore]
src/any_llm/gateway/api/routes/budgets.py:3: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/api/routes/budgets.py:5: error: Cannot find implementation or library stub for module named "sqlalchemy.exc"  [import-not-found]
src/any_llm/gateway/api/routes/budgets.py:6: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/api/routes/budgets.py:51: error: Untyped decorator makes function "create_budget" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/budgets.py:76: error: Untyped decorator makes function "list_budgets" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/budgets.py:88: error: Untyped decorator makes function "get_budget" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/budgets.py:105: error: Untyped decorator makes function "update_budget" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/budgets.py:138: error: Untyped decorator makes function "delete_budget" untyped  [untyped-decorator]
src/any_llm/providers/cerebras/utils.py:3: error: Cannot find implementation or library stub for module named "cerebras.cloud.sdk.types"  [import-not-found]
src/any_llm/providers/cerebras/utils.py:4: error: Cannot find implementation or library stub for module named "cerebras.cloud.sdk.types.chat.chat_completion"  [import-not-found]
src/any_llm/providers/cohere/utils.py:4: error: Cannot find implementation or library stub for module named "cohere"  [import-not-found]
src/any_llm/providers/cohere/utils.py:5: error: Cannot find implementation or library stub for module named "cohere.types"  [import-not-found]
tests/unit/providers/test_together_provider.py:6: error: Cannot find implementation or library stub for module named "together.types"  [import-not-found]
src/any_llm/providers/azure/azure.py:11: error: Cannot find implementation or library stub for module named "azure.ai.inference"  [import-not-found]
src/any_llm/providers/azure/azure.py:12: error: Cannot find implementation or library stub for module named "azure.core.credentials"  [import-not-found]
src/any_llm/providers/azure/azure.py:27: error: Cannot find implementation or library stub for module named "azure.ai.inference.models"  [import-not-found]
src/any_llm/providers/sagemaker/sagemaker.py:18: error: Cannot find implementation or library stub for module named "boto3"  [import-not-found]
src/any_llm/providers/sagemaker/sagemaker.py:107: error: Unused "type: ignore" comment  [unused-ignore]
src/any_llm/gateway/api/routes/pricing.py:3: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/api/routes/pricing.py:5: error: Cannot find implementation or library stub for module named "sqlalchemy.exc"  [import-not-found]
src/any_llm/gateway/api/routes/pricing.py:6: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/api/routes/pricing.py:44: error: Untyped decorator makes function "set_pricing" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/pricing.py:78: error: Untyped decorator makes function "list_pricing" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/pricing.py:90: error: Untyped decorator makes function "get_pricing" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/pricing.py:107: error: Untyped decorator makes function "delete_pricing" untyped  [untyped-decorator]
src/any_llm/providers/bedrock/bedrock.py:19: error: Cannot find implementation or library stub for module named "boto3"  [import-not-found]
src/any_llm/providers/bedrock/bedrock.py:119: error: Unused "type: ignore" comment  [unused-ignore]
tests/gateway/test_log_usage_commit_scope.py:6: error: Cannot find implementation or library stub for module named "sqlalchemy.exc"  [import-not-found]
tests/gateway/test_log_usage_commit_scope.py:7: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/main.py:1: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/main.py:2: error: Cannot find implementation or library stub for module named "fastapi.middleware.cors"  [import-not-found]
src/any_llm/gateway/main.py:3: error: Cannot find implementation or library stub for module named "fastapi.responses"  [import-not-found]
src/any_llm/gateway/main.py:4: error: Cannot find implementation or library stub for module named "starlette.middleware.base"  [import-not-found]
src/any_llm/gateway/main.py:107: error: Class cannot subclass "BaseHTTPMiddleware" (has type "Any")  [misc]
src/any_llm/gateway/main.py:153: error: Untyped decorator makes function "root_tutorial" untyped  [untyped-decorator]
tests/gateway/test_cache_control_headers.py:3: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
tests/gateway/test_cache_control_headers.py:4: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/unit/test_gateway_bootstrap.py:3: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
tests/unit/test_gateway_bootstrap.py:4: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/gateway/conftest.py:11: error: Cannot find implementation or library stub for module named "uvicorn"  [import-not-found]
tests/gateway/conftest.py:12: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
tests/gateway/conftest.py:13: error: Cannot find implementation or library stub for module named "alembic.config"  [import-not-found]
tests/gateway/conftest.py:14: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/conftest.py:15: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
tests/gateway/conftest.py:16: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/gateway/test_usage_tracking.py:7: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_usage_tracking.py:7: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
tests/gateway/test_usage_tracking.py:8: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
tests/gateway/test_usage_tracking.py:9: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/gateway/test_pricing_config.py:6: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_pricing_config.py:7: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/gateway/test_provider_kwargs_override.py:8: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_provider_kwargs_override.py:9: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
tests/gateway/test_provider_kwargs_override.py:10: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
Found 72 errors in 25 files (checked 75 source files)
src/any_llm/gateway/alembic/versions/5911f4bbf98d_set_null_on_user_delete_for_log_.py:11: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
src/any_llm/gateway/alembic/versions/5911f4bbf98d_set_null_on_user_delete_for_log_.py:12: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
src/any_llm/gateway/core/database.py:5: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
src/any_llm/gateway/core/database.py:6: error: Cannot find implementation or library stub for module named "alembic.config"  [import-not-found]
src/any_llm/gateway/core/database.py:7: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
src/any_llm/gateway/core/database.py:8: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/core/database.py:31: error: Untyped decorator makes function "_set_sqlite_pragma" untyped  [untyped-decorator]
src/any_llm/gateway/alembic/versions/10a6a8ead0e7_soft_delete_user_and_cascade_api_key_fk.py:11: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
src/any_llm/gateway/alembic/versions/10a6a8ead0e7_soft_delete_user_and_cascade_api_key_fk.py:12: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
src/any_llm/gateway/alembic/versions/28d153c22616_initial_migration_with_all_models.py:11: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
src/any_llm/gateway/alembic/versions/28d153c22616_initial_migration_with_all_models.py:12: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
src/any_llm/gateway/services/pricing_service.py:3: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/services/pricing_service.py:25: error: Returning Any from function declared to return "ModelPricing | None"  [no-any-return]
src/any_llm/gateway/rate_limit.py:8: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/repositories/users_repository.py:1: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/repositories/users_repository.py:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
src/any_llm/gateway/repositories/users_repository.py:21: error: Returning Any from function declared to return "User | None"  [no-any-return]
src/any_llm/gateway/services/bootstrap_service.py:3: error: Cannot find implementation or library stub for module named "sqlalchemy.exc"  [import-not-found]
src/any_llm/gateway/services/bootstrap_service.py:3: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
src/any_llm/gateway/services/bootstrap_service.py:4: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/repositories/pricing_repository.py:3: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/repositories/pricing_repository.py:25: error: Returning Any from function declared to return "ModelPricing | None"  [no-any-return]
tests/gateway/test_pricing_budget_validation.py:4: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_embeddings_endpoint.py:7: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
src/any_llm/providers/together/utils.py:5: error: Cannot find implementation or library stub for module named "together.types"  [import-not-found]
src/any_llm/providers/gemini/utils.py:8: error: Cannot find implementation or library stub for module named "google.genai"  [import-not-found]
src/any_llm/providers/gemini/utils.py:9: error: Cannot find implementation or library stub for module named "google.genai.pagers"  [import-not-found]
src/any_llm/providers/together/together.py:12: error: Cannot find implementation or library stub for module named "together"  [import-not-found]
src/any_llm/providers/together/together.py:24: error: Cannot find implementation or library stub for module named "together.types"  [import-not-found]
tests/gateway/test_budget_race_condition.py:44: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/providers/cerebras/cerebras.py:13: error: Cannot find implementation or library stub for module named "cerebras.cloud.sdk"  [import-not-found]
src/any_llm/providers/cerebras/cerebras.py:13: error: Cannot find implementation or library stub for module named "cerebras.cloud"  [import-not-found]
src/any_llm/providers/cerebras/cerebras.py:13: error: Cannot find implementation or library stub for module named "cerebras"  [import-not-found]
src/any_llm/providers/cerebras/cerebras.py:14: error: Cannot find implementation or library stub for module named "cerebras.cloud.sdk.types.chat.chat_completion"  [import-not-found]
tests/unit/providers/test_anthropic_provider.py:866: error: Item "ChatCompletionMessageCustomToolCall" of "ChatCompletionMessageFunctionToolCall | ChatCompletionMessageCustomToolCall" has no attribute "function"  [union-attr]
tests/unit/providers/test_anthropic_provider.py:867: error: Item "ChatCompletionMessageCustomToolCall" of "ChatCompletionMessageFunctionToolCall | ChatCompletionMessageCustomToolCall" has no attribute "function"  [union-attr]
tests/unit/providers/test_anthropic_provider.py:869: error: Item "ChatCompletionMessageCustomToolCall" of "ChatCompletionMessageFunctionToolCall | ChatCompletionMessageCustomToolCall" has no attribute "function"  [union-attr]
tests/unit/providers/test_anthropic_provider.py:870: error: Item "ChatCompletionMessageCustomToolCall" of "ChatCompletionMessageFunctionToolCall | ChatCompletionMessageCustomToolCall" has no attribute "function"  [union-attr]
src/any_llm/gateway/api/routes/users.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/api/routes/users.py:6: error: Cannot find implementation or library stub for module named "sqlalchemy.exc"  [import-not-found]
src/any_llm/gateway/api/routes/users.py:7: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/api/routes/users.py:102: error: Untyped decorator makes function "create_user" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/users.py:161: error: Untyped decorator makes function "list_users" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/users.py:173: error: Untyped decorator makes function "get_user" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/users.py:190: error: Untyped decorator makes function "update_user" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/users.py:240: error: Untyped decorator makes function "delete_user" untyped  [untyped-decorator]
src/any_llm/gateway/api/routes/users.py:267: error: Untyped decorator makes function "get_user_usage" untyped  [untyped-decorator]
tests/gateway/test_atomic_spend_update.py:4: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/gateway/test_transaction_safety.py:6: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_transaction_safety.py:7: error: Cannot find implementation or library stub for module named "sqlalchemy.exc"  [import-not-found]
tests/gateway/test_transaction_safety.py:8: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/unit/providers/test_huggingface_provider.py:111: error: Unused "type: ignore" comment  [unused-ignore]
tests/unit/providers/test_huggingface_provider.py:111: error: Cannot find implementation or library stub for module named "huggingface_hub.inference._generated.types"  [import-not-found]
tests/unit/providers/test_huggingface_provider.py:111: note: Error code "import-not-found" not covered by "type: ignore" comment
src/any_llm/gateway/api/routes/embeddings.py:7: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
src/any_llm/gateway/api/routes/embeddings.py:9: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
src/any_llm/gateway/api/routes/embeddings.py:35: error: Untyped decorator makes function "create_embedding" untyped  [untyped-decorator]
tests/unit/providers/test_gemini_provider.py:7: error: Cannot find implementation or library stub for module named "google.genai.types"  [import-not-found]
tests/unit/providers/test_gemini_provider.py:7: error: Cannot find implementation or library stub for module named "google.genai"  [import-not-found]
tests/unit/providers/test_gemini_provider.py:326: error: Value of type "Collection[str]" is not indexable  [index]
tests/unit/providers/test_gemini_provider.py:592: error: Unused "type: ignore" comment  [unused-ignore]
tests/unit/providers/test_gemini_provider.py:593: error: Unused "type: ignore" comment  [unused-ignore]
tests/unit/providers/test_gemini_provider.py:594: error: Unused "type: ignore[index, union-attr]" comment  [unused-ignore]
tests/unit/providers/test_gemini_provider.py:595: error: Unused "type: ignore[union-attr, index]" comment  [unused-ignore]
tests/unit/providers/test_gemini_provider.py:596: error: Unused "type: ignore[union-attr, index]" comment  [unused-ignore]
tests/unit/providers/test_gemini_provider.py:597: error: Unused "type: ignore[union-attr, index]" comment  [unused-ignore]
tests/unit/providers/test_gemini_provider.py:598: error: Unused "type: ignore[union-attr, index]" comment  [unused-ignore]
tests/unit/providers/test_gemini_provider.py:599: error: Unused "type: ignore[union-attr, index]" comment  [unused-ignore]
tests/unit/providers/test_gemini_provider.py:600: error: Unused "type: ignore[union-attr, index]" comment  [unused-ignore]
tests/gateway/test_cors_configuration.py:5: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_cors_configuration.py:6: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/gateway/conftest.py:11: error: Cannot find implementation or library stub for module named "uvicorn"  [import-not-found]
tests/gateway/conftest.py:12: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
tests/gateway/conftest.py:13: error: Cannot find implementation or library stub for module named "alembic.config"  [import-not-found]
tests/gateway/conftest.py:14: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/conftest.py:15: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
tests/gateway/conftest.py:16: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/gateway/test_rate_limiting.py:8: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
tests/gateway/test_rate_limiting.py:9: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_rate_limiting.py:11: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
tests/gateway/test_rate_limiting.py:12: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/gateway/test_metrics.py:8: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
tests/gateway/test_metrics.py:9: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import-not-found]
tests/gateway/test_metrics.py:10: error: Cannot find implementation or library stub for module named "sqlalchemy.orm"  [import-not-found]
tests/gateway/test_budget_reset.py:6: error: Cannot find implementation or library stub for module named "fastapi.testclient"  [import-not-found]
src/any_llm/gateway/cli.py:8: error: Cannot find implementation or library stub for module named "click"  [import-not-found]
src/any_llm/gateway/cli.py:9: error: Cannot find implementation or library stub for module named "uvicorn"  [import-not-found]
src/any_llm/gateway/cli.py:10: error: Cannot find implementation or library stub for module named "uvicorn.config"  [import-not-found]
src/any_llm/gateway/cli.py:17: error: Untyped decorator makes function "cli" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:22: error: Untyped decorator makes function "serve" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:23: error: Untyped decorator makes function "serve" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:30: error: Untyped decorator makes function "serve" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:31: error: Untyped decorator makes function "serve" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:32: error: Untyped decorator makes function "serve" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:33: error: Untyped decorator makes function "serve" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:34: error: Untyped decorator makes function "serve" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:35: error: Untyped decorator makes function "serve" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:36: error: Untyped decorator makes function "serve" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:93: error: Untyped decorator makes function "init_db" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:94: error: Untyped decorator makes function "init_db" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:95: error: Untyped decorator makes function "init_db" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:112: error: Untyped decorator makes function "migrate" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:113: error: Untyped decorator makes function "migrate" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:114: error: Untyped decorator makes function "migrate" untyped  [untyped-decorator]
src/any_llm/gateway/cli.py:115: error: Untyped decorator makes function "migrate" untyped  [untyped-decorator]
Found 102 errors in 29 files (checked 74 source files)

openapi-spec.............................................................Failed
- hook id: openapi-spec
- duration: 1.08s
- exit code: 1

Traceback (most recent call last):
  File "/workspace/.modules/anyllm/scripts/generate_openapi.py", line 17, in <module>
    from any_llm.gateway.core.config import GatewayConfig
  File "/workspace/.modules/anyllm/src/any_llm/gateway/core/config.py", line 8, in <module>
    from pydantic_settings import BaseSettings, SettingsConfigDict
ModuleNotFoundError: No module named 'pydantic_settings'
```

Not sure what is going on, but I should have fixed the errors related to the file I modified.
And I need to run the pre-commit manually, not sure if that is intended.

_Originally posted by @liukidar in https://github.com/mozilla-ai/any-llm/pull/970#issuecomment-4213512374_

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -58,7 +58,12 @@ source .venv/bin/activate
 uv sync --all-extras -U --python=3.13
 
 # 5. Ensure all checks pass
-uv run pre-commit run --all-files --verbose
+UV_SYSTEM_PYTHON=0 uv run pre-commit run --all-files --verbose
+
+> **Note:** mypy errors about missing modules (e.g. `fastapi`, `sqlalchemy`,
+> `groq`, `mistralai`) are expected when running locally without gateway or
+> provider extras installed. These are optional dependencies and CI is the
+> authoritative environment for mypy checks.
 
 # 7. Verify your setup
 pytest -v tests/unit

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -59,16 +59,17 @@ uv sync --all-extras -U --python=3.13
 
 # 5. Ensure all checks pass
 UV_SYSTEM_PYTHON=0 uv run pre-commit run --all-files --verbose
+```
 
 > **Note:** mypy errors about missing modules (e.g. `fastapi`, `sqlalchemy`,
 > `groq`, `mistralai`) are expected when running locally without gateway or
 > provider extras installed. These are optional dependencies and CI is the
 > authoritative environment for mypy checks.
 
+```bash
 # 7. Verify your setup
 pytest -v tests/unit
 pytest -v tests/integration -n auto
-
 ```
 
 ### Setting Up API Keys

